src/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -498,128 +498,8 @@ void MethodContext::recGlobalContext(const MethodContext& other)
     }
 }
 
-void MethodContext::recEnvironment()
-{
-    if (Environment == nullptr)
-        Environment = new DenseLightWeightMap<Agnostic_Environment>();
-
-    char* l_EnvStr;
-    char* l_val;
-
-#ifdef FEATURE_PAL
-    l_EnvStr = GetEnvironmentStringsA();
-#else  // !FEATURE_PAL
-    l_EnvStr = GetEnvironmentStrings();
-#endif // !FEATURE_PAL
-
-    l_val = l_EnvStr;
-
-    char* l_str = l_EnvStr;
-
-    int count = 0;
-    while (true)
-    {
-        if (*l_str == 0)
-            break;
-        while (*l_str != 0)
-            l_str++;
-        l_str++;
-        count++;
-    }
-
-    for (int i = 0; i < count; i++)
-    {
-        if ((_strnicmp(l_EnvStr, "complus_", 8) == 0) || (_strnicmp(l_EnvStr, "dbflag", 6) == 0) ||
-            (_strnicmp(l_EnvStr, "BVT_TEST_ID", 11) == 0))
-        {
-            char* val = l_EnvStr;
-            while (*val != '=')
-                val++;
-            *val++                       = 0;
-            int                  nameind = Environment->AddBuffer((unsigned char*)l_EnvStr, (int)strlen(l_EnvStr) + 1);
-            int                  valind  = Environment->AddBuffer((unsigned char*)val, (int)strlen(val) + 1);
-            Agnostic_Environment value;
-            value.name_index = nameind;
-            value.val_index  = valind;
-            DWORD key        = (DWORD)Environment->GetCount();
-            Environment->Append(value);
-            DEBUG_REC(dmpEnvironment(key, value));
-            l_EnvStr = val;
-        }
-        while (*l_EnvStr != '\0')
-            l_EnvStr++;
-        l_EnvStr++;
-    }
-    FreeEnvironmentStringsA(l_val);
-}
 void MethodContext::dmpEnvironment(DWORD key, const Agnostic_Environment& value)
 {
-    printf("Environment key %u, value '%s' '%s'", key, (LPCSTR)Environment->GetBuffer(value.name_index),
-           (LPCSTR)Environment->GetBuffer(value.val_index));
```

Comment:
I didn't want to invalidate existing MCH files, so I left the LWM definition for `Agnostic_Environment`. The macro that is used to define LWM definitions requires that a `dup` method exists.

I will add a comment to this effect.